### PR TITLE
feat: upgrade fuels-ts to 0.83

### LINKED ISSUE #71: upgrade `fuels-npm-packs` to testnet
- update `fuels-ts` to `0.83.0`

can check how breaking changes were fixed on the [wallet upgrade PR](https://github.com/FuelLabs/fuels-wallet/pull/1240)

[upgrade PR on `fuels-ts`](https://github.com/FuelLabs/fuels-ts/pull/2021/files)

### PR DESCRIPTION
**Pull request by arthurgeron**
Fixes #71 
- Upgrade fuels-ts to 0.83
- Fix error on sequential builds in `test-utils` leading to failure to overwrite type declarations
Screenshot 2024-04-30 at 10 25 52

### CODE REVIEW

File: packages/test-utils/src/test-utils/seedWallet.ts [hunk 1]
```diff
@@ -13,13 +13,14 @@ export async function seedWallet(
   options: SeedWalletOptions = {},
 ) {
   const fuelProvider = await Provider.create(fuelProviderUrl);
-  const { minGasPrice } = await fuelProvider.getGasConfig();
+  const gasPrice = await fuelProvider.getLatestGasPrice();
+  const baseAssetId = fuelProvider.getBaseAssetId();
```
Comment (arthurgeron) [inline packages/test-utils/src/test-utils/seedWallet.ts:17]: @LuizAsFight Is this the expected approach or should we be accepting base asset id as a parameter?

**Review by helciofranco — DISMISSED** [review]
besides that question, LGTM

**Comment by helciofranco** [comment]
looks like the CI is failing

File: packages/test-utils/src/test-utils/seedWallet.ts (hunk 1 above)
Comment (LuizAsFight) [reply packages/test-utils/src/test-utils/seedWallet.ts:17]: about the `baseAssetId` its correct, but about the `gasPrice` I think its not needed as a param anymore for the transfer function, can you check if we can just remove the gasPrice thing?

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

**Review by LuizAsFight — APPROVED** [review]

**Pull request merged**